### PR TITLE
(feat): "Remove Github Thread" context menu

### PR DESCRIPTION
**Pull request by jaw0r3k**
As it is sometimes annoying 
So now u can delete it

### CODE REVIEW

**Comment by MrMythicalYT** [comment]
Why should you use regex on the starter message, instead of checking the channel name? It should be unique to any other threads, right?

**Comment by jaw0r3k** [comment]
> Why should you use regex on the starter message, instead of checking the channel name? It should be unique to any other threads, right?

yh idk

**Comment by JPBM135** [comment]
What happens if the user edits the message?

**Comment by JPBM135** [comment]
I suppose a thread arching emits an event, why not delete the thread when it archives?

**Comment by jaw0r3k** [comment]
> What happens if the user edits the message?

Matching content won't work...
Maybe checking channel name would be better

File: src/commands/removeGithubThread.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import { PermissionFlagsBits } from "discord.js";
+import { NormalGitHubUrlRegex, GitHubUrlLinesRegex } from "../functions/github/regex.js";
+import type { RemoveGithubThread } from "../interactions/context/removeGithubThread.js";
+
+export default class extends Command<typeof RemoveGithubThread> {
+	public constructor() {
+		super(["Remove Github Thread"]);
+	}
+	private matchChannelName(name: string): boolean {
+		return name === "GitHub Lines for this message";
```
Comment (JPBM135) [inline src/commands/removeGithubThread.ts:12]: Maybe move both of the names to a constant?

File: src/commands/removeGithubThread.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	}
+	public override async messageContext(
+		interaction: InteractionParam<CommandMethod.MessageContext>,
+		args: ArgsParam<typeof RemoveGithubThread>,
+	): Promise<void> {
+		if (interaction.channel?.isThread()) {
+			const starterMessage = await interaction.channel.fetchStarterMessage();
+			if (!this.matchChannelName(interaction.channel?.name) && interaction.channel.ownerId !== interaction.client.user.id) {
+				interaction.reply({ content: "This is not github thread.", ephemeral: true });
```
Comment (JPBM135) [inline src/commands/removeGithubThread.ts]:
```suggestion
				await interaction.reply({ content: "This is not github thread.", ephemeral: true });
```

File: src/commands/removeGithubThread.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+				return;
+			}
+			if ( 
+				interaction.user.id !== starterMessage?.author.id &&
+				!interaction.member.permissions.has(PermissionFlagsBits.ManageThreads)
+			) {
+				interaction.reply({ content: "You are not able to do it.", ephemeral: true });
```
Comment (JPBM135) [inline src/commands/removeGithubThread.ts]:
```suggestion
				await interaction.reply({ content: "You are not able to do it.", ephemeral: true });
```

File: src/commands/removeGithubThread.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+				return;
+			}
+			interaction.channel.delete("Remove Github Lines command!");
```
Comment (JPBM135) [inline src/commands/removeGithubThread.ts]: Maybe reply the interaction with `This thread will be deleted in X seconds` and add a timer

```suggestion
			await interaction.channel.delete("Remove Github Lines command!");
```

File: src/commands/removeGithubThread.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+		} else {
+			if (
+				!args.message.hasThread ||
+				args.message.thread?.ownerId !== interaction.client.user.id ||
+				!this.matchChannelName(args.message.thread?.name)
+			) {
+				interaction.reply({ content: "No Github Thread found.", ephemeral: true });
```
Comment (JPBM135) [inline src/commands/removeGithubThread.ts]:
```suggestion
				await interaction.reply({ content: "No Github Thread found.", ephemeral: true });
```

File: src/commands/removeGithubThread.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+				return;
+			}
+			if (
+				interaction.user.id !== args.message?.author.id &&
+				!interaction.member.permissions.has(PermissionFlagsBits.ManageThreads)
+			) {
+				interaction.reply({ content: "You are not able to do it.", ephemeral: true });
```
Comment (JPBM135) [inline src/commands/removeGithubThread.ts]:
```suggestion
				await interaction.reply({ content: "You are not able to do it.", ephemeral: true });
```

File: src/commands/removeGithubThread.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+				return;
+			}
+			args.message.thread?.delete("Remove Github Lines command!");
+			interaction.reply({ content: "Thread succesfully deleted.", ephemeral: true });
```
Comment (JPBM135) [inline src/commands/removeGithubThread.ts]:
```suggestion
			await args.message.thread?.delete("Remove Github Lines command!");
			await interaction.reply({ content: "Thread succesfully deleted.", ephemeral: true });
```

**Review by JPBM135 — CHANGES_REQUESTED** [review]

File: src/commands/removeGithubThread.ts (hunk 4 above)
Comment (jaw0r3k) [reply src/commands/removeGithubThread.ts]: Cool visual  feature 
but useless when using it

Comment (JPBM135) [reply src/commands/removeGithubThread.ts]: The thing is, you will let the interaction fail?

Comment (jaw0r3k) [reply src/commands/removeGithubThread.ts]: But the channel is deleted

**Comment by almostSouji** [comment]
Not liking how chunky context menus become with all these additions. Maybe consider unifying it into a cleanup or similar idea and base what it does on where it is used?

**Comment by jaw0r3k** [comment]
> Not liking how chunky context menus become with all these additions. Maybe consider unifying it into a cleanup or similar idea and base what it does on where it is used?

Maybe connect it with #8 ?

**Comment by JPBM135** [comment]
I think this could move to an automated delete/cleanup only (#7)

**Comment by jaw0r3k** [comment]
Connected with #8

**Pull request closed without merging**